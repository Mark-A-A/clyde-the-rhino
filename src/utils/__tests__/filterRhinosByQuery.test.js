const { filterRhinosByQuery } = require('../filterRhinosByQuery.js');
const rhinoceroses = require('../../data');
describe("filterRhinosByQuery =>", () => {
  it("should return all rhinos if query has no name or species", () => {
    const queryMock = {}
    
    const rhinoList = filterRhinosByQuery(rhinoceroses, queryMock);
    
    expect(rhinoList).toEqual(rhinoceroses);
    expect(rhinoList).toMatchSnapshot();
  });
  
  it("should return specific rhino with name & species", () => {
    const queryMock = {
      "name": "Clyde",
      "species": "white_rhinoceros",
    }

    const rhinoList = filterRhinosByQuery(rhinoceroses, queryMock);
    expect(rhinoList.length).toBe(1);
    
    const [ rhino ] = rhinoList
    expect(rhino.name).toBe("Clyde")
    expect(rhino.species).toBe("white_rhinoceros")
    expect(rhinoList).toMatchSnapshot();
  });

  it("should return all rhinos with specified name ", () => {
    const queryMock = {
      "name": "Winston"
    }

    const rhinoList = filterRhinosByQuery(rhinoceroses, queryMock);
    expect(rhinoList.length).toBe(1);
    
    const [rhino] = rhinoList;
    expect(rhino.name).toBe("Winston")
    
    expect(rhinoList).toMatchSnapshot();
  });

  it("should return all rhinos of a species ", () => {
    const queryMock = {
      "species": "white_rhinoceros"
    }

    const rhinoList = filterRhinosByQuery(rhinoceroses,queryMock);
    
    rhinoList.forEach(rhino => {
      expect(rhino.species).toBe(queryMock.species)
    })

    expect(rhinoList).toMatchSnapshot();
  });
})