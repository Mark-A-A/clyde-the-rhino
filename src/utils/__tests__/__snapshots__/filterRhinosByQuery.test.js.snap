// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterRhinosByQuery => should return all rhinos if query has no name or species 1`] = `
Array [
  Object {
    "id": "08da0048-c5a0-4722-8ec6-75b9b1b046e3",
    "name": "Clyde",
    "species": "white_rhinoceros",
  },
  Object {
    "id": "70314997-b314-4a30-b5bc-1ed9fbd030c8",
    "name": "Clydette",
    "species": "white_rhinoceros",
  },
  Object {
    "id": "3ec7086a-3770-4c41-ac4e-c3139f7b1b5d",
    "name": "Winston",
    "species": "white_rhinoceros",
  },
  Object {
    "id": "8704bbcb-9073-42e4-a3c7-07444dd44bc0",
    "name": "Spike",
    "species": "white_rhinoceros",
  },
  Object {
    "id": "a832bb9e-f20b-4be5-94aa-cf7883939b67",
    "name": "Stompy",
    "species": "javan_rhinoceros",
  },
  Object {
    "id": "10f44082-c158-4a3e-b66d-afc120a1977f",
    "name": "Stompy Jr",
    "species": "javan_rhinoceros",
  },
  Object {
    "id": "d4bc9bfd-80e6-460e-940f-83fcb701d84e",
    "name": "Reginald",
    "species": "javan_rhinoceros",
  },
  Object {
    "id": "cd3f1257-0902-49b5-a9ca-857313228c23",
    "name": "Phil",
    "species": "indian_rhinoceros",
  },
]
`;

exports[`filterRhinosByQuery => should return all rhinos of a species  1`] = `
Array [
  Object {
    "id": "08da0048-c5a0-4722-8ec6-75b9b1b046e3",
    "name": "Clyde",
    "species": "white_rhinoceros",
  },
  Object {
    "id": "70314997-b314-4a30-b5bc-1ed9fbd030c8",
    "name": "Clydette",
    "species": "white_rhinoceros",
  },
  Object {
    "id": "3ec7086a-3770-4c41-ac4e-c3139f7b1b5d",
    "name": "Winston",
    "species": "white_rhinoceros",
  },
  Object {
    "id": "8704bbcb-9073-42e4-a3c7-07444dd44bc0",
    "name": "Spike",
    "species": "white_rhinoceros",
  },
]
`;

exports[`filterRhinosByQuery => should return all rhinos with specified name  1`] = `
Array [
  Object {
    "id": "3ec7086a-3770-4c41-ac4e-c3139f7b1b5d",
    "name": "Winston",
    "species": "white_rhinoceros",
  },
]
`;

exports[`filterRhinosByQuery => should return specific rhino with name & species 1`] = `
Array [
  Object {
    "id": "08da0048-c5a0-4722-8ec6-75b9b1b046e3",
    "name": "Clyde",
    "species": "white_rhinoceros",
  },
]
`;
